### PR TITLE
Integrate with django-pdb.

### PR DESCRIPTION
**Pull request by ryneeverett**
Per issue #483.

I haven't tested very thoroughly (just with the `django-pdb` testapp) but hoping to get some feedback if I'm on the right track.

### CODE REVIEW

**Comment by ryneeverett** [comment]
Ok, as far as I can tell this works just as well as standalone `django-pdb`.

**Comment by trbs** [comment]
@ryneeverett thanks for picking this up !

It looks good, however I would prefer if we wouldn't have django-pdb as a hard requirement.
Could you update the PR so that it passes the flake8 tests and uses django-pdb only if it's available ?

**Comment by ryneeverett** [comment]
The only flake8 errors I'm getting now are for idiomatically importing ipdb and Ipython to see if they're available. I'm assuming it would not be preferable to check `globals()`.

**Comment by ryneeverett** [comment]
Oh wait, that would require it already be imported. Yeah, I'm not aware of any other way to check that.

**Comment by ryneeverett** [comment]
Heh, well I learned something, albeit unuseful.

**Comment by trbs** [comment]
hehe I think easier would be to just add ```# noqa``` at the end of the two import statements :)

**Comment by ryneeverett** [comment]
squashed and green

**Comment by trbs** [comment]
Thanks !

File: django_extensions/management/commands/runserver_plus.py [hunk 1]
```diff
@@ -64,6 +64,13 @@ class Command(BaseCommand):
                     help='auto-reload whenever the given file changes too (can be specified multiple times)'),
         make_option('--reloader-interval', dest='reloader_interval', action="store", type="int", default=DEFAULT_POLLER_RELOADER_INTERVAL,
                     help='After how many seconds auto-reload should scan for updates in poller-mode [default=%s]' % DEFAULT_POLLER_RELOADER_INTERVAL),
+        make_option('--pdb', action='store_true', dest='pdb', default=False,
+                    help='Drop into pdb shell on at the start of any view.'),
```
Comment (blueyed) [inline django_extensions/management/commands/runserver_plus.py]: s/on at/at/

File: django_extensions/management/commands/runserver_plus.py [hunk 2]
```diff
@@ -64,6 +64,13 @@ class Command(BaseCommand):
...
+        make_option('--ipdb', action='store_true', dest='ipdb', default=False,
+                    help='Drop into ipdb shell on at the start of any view.'),
```
Comment (blueyed) [inline django_extensions/management/commands/runserver_plus.py]: s/on at/at/

File: django_extensions/management/commands/runserver_plus.py [hunk 3]
```diff
@@ -149,9 +156,44 @@ def execute(self, sql, params=()):
...
+        try:
+            from django_pdb.middleware import PdbMiddleware
+
+            pdb_option = options.get('pdb', False)
+            ipdb_option = options.get('ipdb', False)
+            pm = options.get('pm', False)
+
+            # Add pdb middleware if --pdb is specified or if in DEBUG mode
+            middleware = 'django_pdb.middleware.PdbMiddleware'
+            if ((pdb_option or settings.DEBUG) and
```
Comment (blueyed) [inline django_extensions/management/commands/runserver_plus.py]: Does this need `or ipdb_option`?

File: django_extensions/management/commands/runserver_plus.py (hunk 1 above)
Comment (ryneeverett) [inline django_extensions/management/commands/runserver_plus.py]: In my defense those typos are in pdb-django too, haha.

File: django_extensions/management/commands/runserver_plus.py (hunk 3 above)
Comment (trbs) [inline django_extensions/management/commands/runserver_plus.py]: I was wondering about that as well

Comment (ryneeverett) [inline django_extensions/management/commands/runserver_plus.py]: Yeah, I couldn't figure out what purpose that served in pdb-django but I forgot to try removing it.

Comment (blueyed) [inline django_extensions/management/commands/runserver_plus.py]: Without it, --ipdb would not add the middleware?!

Comment (ryneeverett) [inline django_extensions/management/commands/runserver_plus.py]: I guess the idea is that you can use pdb-django when `DEBUG=False` if you explicitly pass a flag, but the other ways of entering a debugger are disallowed.

Comment (ryneeverett) [inline django_extensions/management/commands/runserver_plus.py]: > Without it, --ipdb would not add the middleware?!

Hmm?

Comment (blueyed) [inline django_extensions/management/commands/runserver_plus.py]: Yes, you can explicitly use `--pdb`, but not `--ipdb` currently.

It should be:

    if ((pdb_option or ipdb_option or settings.DEBUG) and

Or move it down and use `settings.DEBUG or PdbMiddleware.always_break``.

Comment (ryneeverett) [inline django_extensions/management/commands/runserver_plus.py]: Oh wow, good point.

Comment (trbs) [inline django_extensions/management/commands/runserver_plus.py]: committed the first option

**Pull request merged**